samples: benchmark: coremark: Add nrf54h20dk support (#15382)
nrfconnect/sdk-nrf: samples/benchmarks/coremark/README.rst

   It also disables LEDs and buttons.
   Otherwise, it will wait for the button to be pressed.

.. note::
   For the radio core and the PPR core the :kconfig:option:`CONFIG_APP_MODE_FLASH_AND_RUN` Kconfig option is always enabled.
   The buttons are not used for these cores.

.. _SB_CONFIG_APP_CPUNET_RUN:

SB_CONFIG_APP_CPUNET_RUN - Enable execution for the network core or the radio core
   Enable the benchmark execution also for the network core for targets with the nRF53 Series SoCs or for the radio core for the targets with the nRF54H20 SOCs.

.. _SB_CONFIG_APP_CPUPPR_RUN:

SB_CONFIG_APP_CPUPPR_RUN - Enable execution for the PPR core
   Enable the benchmark execution also for the PPR core for targets with the nRF54H20 SOCs.

.. note::
   PPR code is run from MRAM with the execution in place (XIP) method.
   You must use the "nordic-ppr-xip" snippet for the Application Core to be able to boot the PPR core.

>    You can do it by providing the following build argument: `coremark_SNIPPET=nordic-ppr-xip`.

Reviewer: Add also a full build command as an example

Author: I can add:
`west build -b nrf54h20dk/nrf54h20/cpuapp --sysbuild -- -DSB_CONFIG_APP_CPUNET_RUN=n -DSB_CONFIG_APP_CPUPPR_RUN=y -Dcoremark_SNIPPET=nordic-ppr-xip`
or without the Kconfig options:
`west build -b nrf54h20dk/nrf54h20/cpuapp --sysbuild --pristine -- -Dcoremark_SNIPPET=nordic-ppr-xip`
What do you think would be better?

Reviewer: If both options work, I would pick the simpler one

   It also disables LEDs and buttons.

>    Otherwise, it will wait for the button to be pressed.

Reviewer 2: ```suggestion
   Otherwise, it will wait for the button press.
```

.. note::
   For the radio core and the PPR core the :kconfig:option:`CONFIG_APP_MODE_FLASH_AND_RUN` Kconfig option is always enabled.
   The buttons are not used for these cores.

.. _SB_CONFIG_APP_CPUNET_RUN:

SB_CONFIG_APP_CPUNET_RUN - Enable execution for the network core or the radio core

>    Enable the benchmark execution also for the network core for targets with the nRF53 Series SoCs or for the radio core for the targets with the nRF54H20 SOCs.

Reviewer 2: ```suggestion
   Enable the benchmark execution for the network core for targets with the nRF53 Series SoCs, and for the radio core on targets with the nRF54H20 SoCs.
```

.. _SB_CONFIG_APP_CPUPPR_RUN:

SB_CONFIG_APP_CPUPPR_RUN - Enable execution for the PPR core
   Enable the benchmark execution also for the PPR core for targets with the nRF54H20 SOCs.

.. note::
   PPR code is run from MRAM with the execution in place (XIP) method.
   You must use the "nordic-ppr-xip" snippet for the Application Core to be able to boot the PPR core.

>    You can do it by providing the following build argument: `coremark_SNIPPET=nordic-ppr-xip`.

Reviewer 3: ```suggestion
   Use the build argument ``coremark_SNIPPET=nordic-ppr-xip``.
```
